test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,178 @@
+	mustLoadImages string
+)
+
+// Test suite global vars
+var (
+	// clusterProvider manages provisioning of the the bootstrap cluster to be used for the e2e tests.
+	// Please note that provisioning will be skipped if e2e.use-existing-cluster is provided.
+	clusterProvider bootstrap.ClusterProvider
+
+	// clusterProxy allows to interact with the kind cluster to be used for the e2e tests.
+	clusterProxy framework.ClusterProxy
+)
+
+const csiNamespace = "secrets-store-csi-driver"
+
+func init() {
+	flag.BoolVar(&skipCleanup, "e2e.skip-resource-cleanup", false, "if true, the resource cleanup after tests will be skipped")
+	flag.StringVar(&nodeImage, "e2e.node-image", "", "kindest/node image tag")
+	flag.StringVar(&chartPath, "e2e.chart-path", "", "helm chart path")
+	flag.StringVar(&manifestsDir, "e2e.manifests-dir", "", "manifests directory path")
+	flag.StringVar(&kindconfigPath, "e2e.kindconfig-path", "", "kind configuration file path")
+	flag.StringVar(&tryLoadImages, "e2e.try-load-images", "", "space separated list of image tag if exists in local")
+	flag.StringVar(&mustLoadImages, "e2e.must-load-images", "", "space separated list of image tag that must be in local")
+}
+
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "secrets-store-csi-driver")
+}
+
+var _ = BeforeSuite(func() {
+	By("Setting up the cluster")
+	clusterProvider, clusterProxy = setupCluster(initScheme())
+
+	ctx := context.TODO()
+
+	By("Install CSI driver")
+	cli := clusterProxy.GetClient()
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: csiNamespace,
+		},
+	}
+	Eventually(func() error {
+		return cli.Create(ctx, ns)
+	}, framework.CreateTimeout, framework.CreatePolling).Should(Succeed())
+
+	csidriver.InstallAndWait(ctx, csidriver.InstallAndWaitInput{
+		Getter:         cli,
+		KubeConfigPath: clusterProxy.GetKubeconfigPath(),
+		ChartPath:      chartPath,
+		Namespace:      csiNamespace,
+	})
+})
+
+var _ = AfterSuite(func() {
+	if !skipCleanup {
+		By("Tearing down the kind cluster")
+		tearDown(clusterProvider, clusterProxy)
+	}
+})
+
+func initScheme() *runtime.Scheme {
+	sc := runtime.NewScheme()
+
+	_ = corev1.AddToScheme(sc)
+	_ = appsv1.AddToScheme(sc)
+	_ = rbacv1.AddToScheme(sc)
+	_ = v1alpha1.AddToScheme(sc)
+
+	return sc
+}
+
+func setupCluster(scheme *runtime.Scheme) (bootstrap.ClusterProvider, framework.ClusterProxy) {
+	var clusterProvider bootstrap.ClusterProvider
+	kubeconfigPath := ""
+
+	images := []framework.ContainerImage{}
+	for _, c := range strings.Split(tryLoadImages, " ") {
+		images = append(images, framework.ContainerImage{
+			Name:         c,
+			LoadBehavior: framework.TryLoadImage,
+		})
+	}
+	for _, c := range strings.Split(mustLoadImages, " ") {
+		images = append(images, framework.ContainerImage{
+			Name:         c,
+			LoadBehavior: framework.MustLoadImage,
+		})
+	}
+
+	clusterProvider = bootstrap.CreateKindClusterAndLoadImages(context.TODO(), bootstrap.CreateKindClusterAndLoadImagesInput{
```

Comment:
I really like how easy it makes it to run the tests locally (particularly beginner friendly), and the fact it gives you a completely fresh cluster with no state pollution.

Can we make cluster creation optional via an env var or similar? Preferably the default should be to create a fresh cluster, because CI is the main place we might want to disable it, and there's ~0 overhead to setting an extra env var in CI.

On a semi-relevant note, when I tried the bats tests, I found they weren't compatible (without modification) with `minikube` because the TLS config was different to `kind`. I haven't tried these tests with `minikube`, but as a general comment we shouldn't assume they'll work with anything other than what we test them with - we should probably mention the test pre-requisites in the readme, whatever those end up being.